saleor/graphql/tax/dataloaders.py

```diff
@@ -57,3 +60,24 @@ def batch_load(self, keys):
             keys
         )
         return [tax_class_map.get(obj_id) for obj_id in keys]
+
+
+class ProductChargeTaxesByTaxClassIdLoader(DataLoader):
+    context_key = "product_charge_taxes_by_tax_class_id"
+
+    def batch_load(self, keys):
+        non_zero_rates = TaxClassCountryRate.objects.filter(
+            tax_class=OuterRef("pk")
+        ).exclude(rate=Decimal(0))
+        tax_class_map = (
+            TaxClass.objects.using(self.database_connection_name)
+            .filter(pk__in=keys)
+            .annotate(charge_taxes=Exists(non_zero_rates))
+            .in_bulk(keys)
+        )
+        return [
+            tax_class_map[tax_class_id].charge_taxes
+            if tax_class_map.get(tax_class_id)
+            else False
```

Comment:
Shouldn't we return "None" in this case? When I pass some fake/invalid key to this dataloader I receive `False` as a response because such `TaxClass`   or `TaxClassCountryRate` don't exist.

Reply:
This behavior is added only for deprecated fields `Product.chargeTaxes` and `ProductType.chargeTaxes` and only relies on existing tax classes to determine if there are any non-zero rates set. If there is no tax class, there is no chance that there is any tax class, so I think returning `False` is reasonable value. I added a note about this deprecated behavior to the dataloader, so that it isn't used in any other cases.